### PR TITLE
[WIP] MAILBOX-372 RabbitMQEventBus timeout handling

### PR DESCRIPTION
**Pull request by trantienduchn**
other stuffs:
- timeout handling should be a part of the contract, therefore, have to implement InVM timeout handling
- need a timeout configuration property
- write a test of the scenario when rabbitMQ crash (I don't know how to setup this test honestly, need some helps)

### CODE REVIEW

**Comment by chibenwa** [comment]
`timeout handling should be a part of the contract, therefore, have to implement InVM timeout handling`

No. A timeout `In VM` is a non sens as it is linked to a James node failure. Then InVMEventBus should not enforce timeout - inherents to distributed constraints.

`need a timeout configuration property` -> Only for RabbitMQEventBus

`write a test of the scenario when rabbitMQ crash (I don't know how to setup this test honestly, need some helps)`

 - 1. start 2 eventBus
 - 2. Distatch an event
 - 3. lock the execution of the group using a CountLatchDown
 - 4. Stop the EventBus that executed that group
 - 5. The event should then be proceessed by the remaining EventBus

File: mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/GroupRegistration.java [hunk 1]
```diff
@@ -149,13 +152,20 @@ private void subscribeWorkQueue() {
         int currentRetryCount = getRetryCount(acknowledgableDelivery);
 
         return delayGenerator.delayIfHaveTo(currentRetryCount)
-            .flatMap(any -> Mono.fromRunnable(() -> mailboxListener.event(event)))
+            .flatMap(any -> deliverToListenerOnDedicatedScheduler(event))
+            .timeout(DEFAULT_CONSUMER_TIMEOUT)
```
[the remark this answers is not in the document]
Comment (trantienduchn) [reply mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/GroupRegistration.java]: I think now I understand want you want. Then I think this PR is useless. As I searched in the morning there is no timeout configuration option for rabbitMQ, it waits forever.

**Comment by chibenwa** [comment]
Read https://www.rabbitmq.com/confirms.html

**Comment by trantienduchn** [comment]
see #2110

**Pull request closed without merging**